File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/word_in_collection/response/word_in_collection_response_dto.js
import {
    CustomerCollectionResponseDTO
} from "../../customer_collection/response/customer_collection_response_dto.js";

import {
    WordResponseDTO
} from "../../word/response/word_response_dto.js";

export class WordInCollectionResponseDTO {
    #id;
    #dateOfAdditional;
    #customerCollection;
    #word;

    constructor(json) {
        if (json) {
            this.#id = json["id"];

            let dateOfAdditional = json["date_of_additional"];
            if (dateOfAdditional) {
                this.#dateOfAdditional = new Date(dateOfAdditional);
            }

            let customerCollection = json["customer_collection"];
            if (customerCollection) {
                this.#customerCollection = new CustomerCollectionResponseDTO(customerCollection);
            }

            let word = json["word"];
            if (word) {
                this.#word = new WordResponseDTO(word);
            }
        }
    }

    getId() {
        return this.#id;
    }

    getDateOfAdditional() {
        return this.#dateOfAdditional;
    }

    getCustomerCollection() {
        return this.#customerCollection;
    }

    getWord() {
        return this.#word;
    }
}

File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/wordincollection/response/WordInCollectionResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.wordincollection.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.controllers.json.l.LongDeserializer;
import ru.herooo.mylanguageweb.controllers.json.l.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.customercollection.response.CustomerCollectionResponseDTO;
import ru.herooo.mylanguageweb.dto.entity.word.response.WordResponseDTO;

import java.time.LocalDateTime;

public class WordInCollectionResponseDTO {
    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("id")
    private long id;

    @JsonProperty("date_of_additional")
    private LocalDateTime dateOfAdditional;

    @JsonProperty("customer_collection")
    private CustomerCollectionResponseDTO customerCollection;

    @JsonProperty("word")
    private WordResponseDTO word;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public LocalDateTime getDateOfAdditional() {
        return dateOfAdditional;
    }

    public void setDateOfAdditional(LocalDateTime dateOfAdditional) {
        this.dateOfAdditional = dateOfAdditional;
    }

    public CustomerCollectionResponseDTO getCustomerCollection() {
        return customerCollection;
    }

    public void setCustomerCollection(CustomerCollectionResponseDTO customerCollection) {
        this.customerCollection = customerCollection;
    }

    public WordResponseDTO getWord() {
        return word;
    }

    public void setWord(WordResponseDTO word) {
        this.word = word;
    }
}
